libr/util/diff.c

```diff
@@ -358,67 +358,50 @@ R_API bool r_diff_buffers_distance_myers(RDiff *diff, const ut8 *a, ut32 la, con
 	return true;
 }
 
-R_API bool r_diff_buffers_distance_original(RDiff *d, const ut8 *a, ut32 la, const ut8 *b, ut32 lb, ut32 *distance, double *similarity) {
-	int i, j, tmin, **m;
-	ut64 totalsz = 0;
-
-	if (!a || !b || la < 1 || lb < 1)
+R_API bool r_diff_buffers_distance_original(RDiff *diff, const ut8 *a, ut32 la, const ut8 *b, ut32 lb, ut32 *distance, double *similarity) {
+	if (!a || !b)
 		return false;
 
-	if (la == lb && !memcmp (a, b, la)) {
```

Comment:
why remove that optimization?

Reply:
It's fully included in the stripping prefix/suffix optimization.

Reply:
wat, i dont see you comparing the pointers

Reply:
```c
	// Strip prefix
	for (; a < ea && b < eb && *a == *b; a++, b++) {}
	// Strip suffix
	for (; a < ea && b < eb && ea[-1] == eb[-1]; ea--, eb--) {}
```

If `a` and `b` are the same, the following stuff will take very little time so I don't bother add the `if (!la && !lb)` fast path.